### PR TITLE
LU-3407 first frame

### PR DESCRIPTION
**Pull request by animanathome**
When seeking to time 0 for our avatar `webm` videos, we wouldn't be able a frame. That's because we would skip a packet if its PTS was larger than the target PTS. So if we're looking for a PTS of 0 and the first packet would have a PTS of 100 then we would return (and thus end up with nothing). I've fixed this here so if the first packet has a higher PTS then the requested PTS, then we use that one which solves our no frame at 0 issue.

### CODE REVIEW

File: src/backends/beamcoder.ts [hunk 1]
```diff
@@ -390,19 +391,22 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
...
                 filteredFrames = filteredResult.flatMap(r => r.frames);
-                VERBOSE && console.log('filteredFrames', filteredFrames.length, 'filteredFrames.pts:', filteredFrames.map(f => f.pts), '-> target.pts:', targetPTS);
+                VERBOSE && console.log('filteredFrames', filteredFrames.length, 'filteredFrames.pts:', JSON.stringify(filteredFrames.map(f => f.pts)), '-> target.pts:', targetPTS);
 
                 // get the closest frame to our target presentation timestamp (PTS)
                 // Beamcoder returns decoded packet frames as follows: [1000, 2000, 3000, 4000]
+                // If we're looking for a frame at 0, we want to return the frame at 1000
                 // If we're looking for a frame at 2500, we want to return the frame at 2000
-                const closestFrame = filteredFrames.reverse().find(f => f.pts <= targetPTS);
+                const closestFrame = (this.#packetReadCount === 1 && filteredFrames[0].pts > targetPTS)
```
Comment (michaeljherrmann) [inline src/backends/beamcoder.ts:407]: will this work with looping too? like if we seek back to an earlier timestamp. I'm not sure how framefusion handles that but could be a case to consider? So if we play and then loop back to the start (maybe not a use case for AI avatar specifically now but is a use case more generally)

**Review by michaeljherrmann — APPROVED** [review]

File: src/backends/beamcoder.ts (hunk 1 above)
Comment (animanathome) [reply src/backends/beamcoder.ts:407]: I'll add a test for that

Comment (animanathome) [reply src/backends/beamcoder.ts:407]: So we have a few tests for looping such as `can get frames at random times (forward and backward)` and `can get frames when looping`.

Comment (michaeljherrmann) [reply src/backends/beamcoder.ts:407]: but I'm guessing those tests don't use the new code path? Like the case for when PTS is greater than what we're looking for

**Pull request merged**